noisyatom/blogs/models.py

```diff
@@ -0,0 +1,90 @@
+
+from django.conf import settings
+from django.contrib.contenttypes.models import ContentType
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.db.models.signals import pre_save
+from django.utils import timezone
+from django.utils.safestring import mark_safe
+from django.utils.text import slugify
+
+from markdown_deux import markdown
+
+
+'''
+https://docs.djangoproject.com/en/1.11/ref/models/fields/
+'''
+
+class PostManager(models.Manager):
+
+	def active(self, *args, **kwargs):
+		return super(PostManager, self).filter(
+										draft=False).filter(
+													publish__lte=timezone.now())
+
+
+def upload_image_location(instcance, filename):
+	return '%s/%s'%(instcance.id, filename)
+
+
+class Post(models.Model):
+	user = models.ForeignKey(settings.AUTH_USER_MODEL, default=1)
+	title = models.CharField(max_length=200)
+	slug = models.SlugField(unique=True)
+	content = models.TextField()
+	image = models.ImageField(upload_to=upload_image_location, 
+								null=True, blank=True, 
+								height_field='height_field',
+								width_field='width_field')
+	height_field = models.IntegerField(default=0)
+	width_field = models.IntegerField(default=0)
+
+	draft = models.BooleanField(default=False)
+	publish = models.DateField(auto_now=False, auto_now_add=False)
+
+	updated = models.DateTimeField(auto_now=False, auto_now_add=True)
+	created = models.DateTimeField(auto_now=True, auto_now_add=False)
+
+	objects = PostManager()
+
+	def __str__(self):
+		return self.title
+
+	def get_absolute_url(self):
+		return reverse('blogs:detail', kwargs={'slug': self.slug})
+
+	class Meta:
+		ordering = ['-created', '-updated']
+
+	def get_markdown(self):
+		content = self.content
+		markdown_text = markdown(content)
+		return mark_safe(markdown_text)
+
+
+	@property
+	def get_content_type(self):
+		instance = self
+		content_type = ContentType.objects.get_for_model(instance.__class__)
+		return content_type
+
+
+def create_slug(instance, new_slug=None):
+	slug = slugify(instance.title)
+	if new_slug is not None:
+		slug = new_slug
+	qs = Post.objects.filter(slug=slug).order_by('-id')
+	exists = qs.exists()
+	# is the title exist than create new title via slug
+	if exists:
+		new_slug = '%s-%s' %(slug, qs.first().id)
+		return create_slug(instance, new_slug=new_slug)
+	return slug
+'''
+that is mean before the model will be save to the db we will do something!
```

Comment:
Can you put a better doc string in this please?

Reply:
Also the doc string is not indented.

Reply:
done